controllers/service_controller.go

```diff
@@ -119,24 +119,34 @@ func (r *ServiceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		namespace = req.Name
 		n := &corev1.Namespace{}
 		if err := r.Client.Get(ctx, req.NamespacedName, n); err != nil {
+			if k8serr.IsNotFound(err) {
```

Comment:
Previously, controller keep checking namespaces even though the namespaces do not exist anymore. 
As a result,  a lot of misleading error messages showed up in the log. (I don't find the snippet of the message now)

Reply:
@tjohnson31415 Thanks for the review.  From my 2 cents, the operator keeps watching the namespace so I don't think there is a possibility of missing the terminated status of the Namespace object. When any namespace changes its status, this operator detects it. WDYT?

Regarding the `IsNotFound` part, do you want me to update this pr without it? and send another pr to update it when this original PR is merged?

Reply:
one more comment about the namespace object status. The main issue was the operator does not have logic when the namespace is under `Terminating` status([source](https://github.com/kserve/modelmesh-serving/blob/main/controllers/util.go#L26)).  From my understanding,  `Terminating` is the only problematic status in this procedure
- `Active` + ` modelmesh-enabled"` label ==> modelMeshEnabled return true
- `Terminating` + ` modelmesh-enabled"` label ==> modelMeshEnabled return true
- `Active` + no label ==>  modelMeshEnabled return false
The other status does not impact this logic.

The most important point here is, that when any namespace's status changes, this [reconclie](https://github.com/kserve/modelmesh-serving/blob/main/controllers/service_controller.go#L112) will be called and check the above logic. When any namespace gets `terminate` signal, the status will be changed to `Terminating`. 

please correct me if something is wrong here.

Reply:
In the "happy path" case, I think that reconciliation should always be able to observe the terminating state. In error/edge cases (eg. many namespaces deleted in bulk, error encountered during reconciliation and a requeue happening), I'm not so sure since "Terminating" is a temporary status.

Before this PR, the reconciliation handled neither "Terminating" nor "Not Found". This PR implements clean-up if the namespace is seen as Terminating, but, If reconciliation happens for a namespace that no longer exists, should the code:
(a) Assume that in-memory resources for that namespace do not exists and just skip reconciliation (i.e. assume that if modelmesh was enabled for that namespace that it previously reconciled while it was Terminating)
(b) Check for in-memory resources that exist and clean them up if they do

With the "level-based" reconciliation that Operators are meant to implement, I think (b) makes more sense as it reconciles based on the current state, without assumption of past reconciliation.